core/src/main/java/org/jruby/javasupport/Java.java

```diff
@@ -521,7 +521,8 @@ else if ( clazz == Object.class ) {
             for ( int i = interfaces.length; --i >= 0; ) {
                 proxy.includeModule(getInterfaceModule(runtime, interfaces[i]));
             }
-            if ( Modifier.isPublic(clazz.getModifiers()) ) {
+            // we don't want to expose JRuby internal generated classes such as org.jruby.gen.InterfaceImpl
+            if (Modifier.isPublic(clazz.getModifiers()) && !clazz.isSynthetic() && !Proxy.isProxyClass(clazz)) {
```

Comment:
I think the comment should be more general... we don't want to expose *any* synthetic classes.

Reply:
changed it to: https://github.com/jruby/jruby/pull/8503/files#diff-41d5e3a18474c6013321ccd78b4084effc7f6504032f053413df31c227f76720R524-R525

also removed the `isProxyClass` check, that felt risky in terms of causing regressions. 
was originally motivated by the fact that there's `-Xinterfaces.useProxy` to switch to using `java.lang.reflect.Proxy` for interface impls but I do not believe it's used much (actually seems broken and should maybe be removed in 10).

Reply:
Ok sounds good to me.